File: erpnext/accounts/report/journal_entry_list/journal_entry_list.py
import frappe, json
from frappe import _

def execute(filters=None):
    # set filters
    order = ""
    cond = ""
    if filters.get("order_by") == "Date":
        order = "a.posting_date DESC"
    else:
        order = "a.name DESC"

    if filters.get("name"):
        cond += " and a.name like '%"+filters.get("name")+"%' "


    # get column
    column = [
        {"label": _("Name"),            "fieldtype":"Data",    "fieldname": "name",             "width": 180},
		{"label": _("Date"),            "fieldtype":"Date",    "fieldname": "posting_date",     "width": 100},
        {"label": _("Status"),          "fieldtype":"Data",    "fieldname": "docstatus",        "width": 100},
        {"label": _("Entry Type"),      "fieldtype":"Data",    "fieldname": "voucher_type",     "width": 110},
        {"label": _("Company"),         "fieldtype":"Data",    "fieldname": "company",          "width": 150},
        {"label": _("Reference Number"),"fieldtype":"Data",    "fieldname": "cheque_no",      "width": 150},
        {"label": _("Total Debit"),     "fieldtype":"Currency",    "fieldname": "total_debit",      "width": 120},
        {"label": _("Remark"),     "fieldtype":"Data",    "fieldname": "remark",     "width": 200}
    ]

    # get data
    data = frappe.db.sql("""
        SELECT 
            a.*
        FROM
            (SELECT 
                j.name,
                    j.posting_date,
                    IF(j.docstatus = 0, 'Draft', IF(j.docstatus = 1, 'Submitted', 'Cancelled')) as docstatus,
                    "{{}}" as data,
                    j.voucher_type,
                    j.company,
                    j.cheque_no,
                    j.total_debit,
                    j.remark         
            FROM
                `tabJournal Entry` j UNION ALL SELECT 
                    d.deleted_name,
                    d.document_date,
                    'Deleted' AS docstatus,
                    d.data, d.name, d.name,d.name,d.name,d.name
            FROM
                `tabDeleted Document` d
            WHERE
                d.deleted_doctype = 'Journal Entry') a
        where a.name is not null {} 
                         ORDER BY {}
    """.format(cond, order), as_dict=1)

    # process
    final_data = []
    for d in data:
        if d.docstatus == "Deleted":
            dt = json.loads(d.data)
            dt["docstatus"] = "Deleted"
            final_data.append(dt)
        else:
            final_data.append(d)
    return column, final_data

File: erpnext/accounts/report/journal_entry_list/journal_entry_list.js
// Copyright (c) 2024, Frappe Technologies Pvt. Ltd. and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Journal Entry List"] = {
	"filters": [
		{
			"fieldname":"order_by",
			"label": __("Order By"),
			"fieldtype": "Select",
			"options": "Name\nDate",
			"default":"Date"
		},
		{
			"fieldname":"name",
			"label": __("ID"),
			"fieldtype": "Data",
			"width": 100,
		},
		// {
		// 	"fieldname":"period_start_date",
		// 	"label": __("Start Date"),
		// 	"fieldtype": "Date",
		// 	"reqd": 1,
		// 	"default": frappe.datetime.add_months(frappe.datetime.get_today(), -1)
		// },
		// {
		// 	"fieldname":"period_end_date",
		// 	"label": __("End Date"),
		// 	"fieldtype": "Date",
		// 	"reqd": 1,
		// 	"default": frappe.datetime.get_today()
		// },		
	]
};
